dart-mode.el

```diff
@@ -562,6 +579,251 @@ This uses `dart-format-path' to find the formatter."
       (dart-format-region beg (point)))))
 
 
+;;; Dart analysis server
+
+(defvar dart-debug nil)
+(defvar dart-enable-analysis-server nil)
+(defvar dart-analysis-server nil)
+(defvar dart-analysis-server-snapshot-path
+  "/usr/lib/dart/bin/snapshots/analysis_server.dart.snapshot")
```

Comment:
This is not a portable path. It should default to `(concat (file-name-directory dart-executable-path) (file-name-as-directory "snapshots") "analysis_server.dart.snapshot")`.